packages/cinterop/src/darwin/kotlin/io/realm/internal/interop/sync/SyncErrorCodeCategory.kt

```diff
@@ -0,0 +1,22 @@
+package io.realm.internal.interop.sync
+
+import realm_wrapper.realm_sync_error_category
+
+actual enum class SyncErrorCodeCategory(val nativeValue: realm_sync_error_category) {
+    CLIENT(realm_sync_error_category.RLM_SYNC_ERROR_CATEGORY_CLIENT),
+    CONNECTION(realm_sync_error_category.RLM_SYNC_ERROR_CATEGORY_CONNECTION),
+    SESSION(realm_sync_error_category.RLM_SYNC_ERROR_CATEGORY_SESSION),
+    SYSTEM(realm_sync_error_category.RLM_SYNC_ERROR_CATEGORY_SYSTEM),
+    UNKNOWN(realm_sync_error_category.RLM_SYNC_ERROR_CATEGORY_UNKNOWN);
+
+    companion object {
+        fun of(nativeValue: realm_sync_error_category): SyncErrorCodeCategory {
+            for (value in values()) {
+                if (value.nativeValue == nativeValue) {
+                    return value
+                }
+            }
+            error("Unknown sync error code category: $nativeValue")
```

Comment:
Won't this reintroduce the breaking change we're trying to avoid when the protocol changes or introduce a new error code?

Reply:
Currently the category is not exposed to end users, but gets serialized to a String, so we can change it without breaking people. But I expect we are going to revisit this once we have a better idea about how to expose errors. Right now, the shortest path was just to expose what the C-API has.

But we serialize all of it to a String, which is also pretty bad, but is what we have done so far.

This will be fixed in https://github.com/realm/realm-kotlin/issues/524